bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/database/ShadeCapabilitiesDatabase.java

```diff
@@ -58,18 +59,20 @@ public class ShadeCapabilitiesDatabase {
      */
     private static final Map<Integer, Type> TYPE_DATABASE = Arrays.asList(
     // @formatter:off
+            new Type( 1).capabilities(0).text("Roller / Solar"),
             new Type( 4).capabilities(0).text("Roman"),
             new Type( 5).capabilities(0).text("Bottom Up"),
             new Type( 6).capabilities(0).text("Duette"),
             new Type( 7).capabilities(6).text("Top Down"),
             new Type( 8).capabilities(7).text("Duette Top Down Bottom Up"),
             new Type( 9).capabilities(7).text("Duette DuoLite Top Down Bottom Up"),
-            new Type(18).capabilities(1).text("Silhouette"),
+            new Type(18).capabilities(1).text("Pirouette"),
             new Type(23).capabilities(1).text("Silhouette"),
             new Type(38).capabilities(9).text("Silhouette Duolite"),
             new Type(42).capabilities(0).text("M25T Roller Blind"),
             new Type(43).capabilities(1).text("Facette"),
-            new Type(44).capabilities(0).text("Twist"),
+            new Type(44).capabilities(0).text("Twist")
+                        .comment("Shade has functionality of a capabilities 1 shade").typeCapabilities(1),
```

Comment:
Technically Type 44 (Twist) shades report themselves as Capabilities 0, which technically means that they are simple roller blinds, without any vanes. But as we both know, this type of shade does actually emulate a vane/tilt function by slightly moving the overlapping bands to cover / not cover each other; and they do therefore also support the OH binding vane/tilt functionality. In other words, they report no vane/tilt capability even though they do actually support it. So this `typeCapabilities()` attribute is used to tell the binding: _"even if the blind technically reports a certain capabilities level, please actually use this other typeCapabilities level instead"_.

Reply:
So difference compared to simply declaring `capabilities(1)` is that you use text from capabilities 0 (visible as property), but attributes from capabilities 1?

Reply:
Yes.

Reply:
OK, finally I had time to understand it from your code and explanation, so basically a work-around to override capabilities when wrongly reported by the API. I see now from returned JSON as you wrote:
```
"type":44,
"capabilities":0
```

So indeed inconsistent. I checked also the code you have provided for now looking up capabilities in internal database first, and only falling back to capabilities provided by JSON response if not found. Looks good. The only thing nagging me is that capabilities in database overrides reported capabilities - but also typeCapabilities overrides the already overridden capabilities. Order of precedence:
- typeCapabilities in database (if present) from type
- capabilities in database (if present) from type
- capabilities in response

So some complexity introduced only to provide a different text for the overridden capabilities. But decoupling texts from capabilities in itself would also add complexity, so I don't have a good proposal right now. At least this is only for Twist. Resolving comment.

Reply:
I hear what you say about the complexity. The capabilities and types database is a kind of internal record of the state of knowledge about shade types and how they function. For example it is used for testing the binding functionality for different types in the JUnit `ShadePositionTests` class, so the tests can be done without the need for developers to physically own an example of all the different types of shades. And when the binding is online, it reads the `type` and `capabilities` reported by the shade and cross checks these against what is in the database. And if there is any kind of discrepancy the binding produces a warning in the log to contact us (the developers). This will allow us to expand our knowledge over time whenever new types of shades and/or new capabilities are discovered, and update the code accordingly.